### PR TITLE
Rotate hexagonal assembly patches correctly for the hex grid type

### LINKED ISSUE #1878: `insertCoreAndAssemblyMaps()` is broken after #1649
I believe that #1649 broke the plotting functionality in [`insertCoreAndAssemblyMaps()`](https://github.com/terrapower/armi/blob/abd1db4adfa0c536f9bf9f08f0f1e8b74b1cd0d7/armi/bookkeeping/report/newReportUtils.py#L617). Specifically, the rotation of assembly patches is not in line with the rotation of the core. Here is an example for the initial core of FFTF (public information):

Before #1649 :
fftfmc_P1P0_MC2v3_20cm_666dRadialCoreMap

After #1649 :
fftfmc_P1P0_MC2v3_20cm_666dRadialCoreMap

### PR DESCRIPTION
**Pull request by keckler**
## What is the change?

The patches in hexagonal facemap plots were always being rotated by pi/2. This should only happen conditionally on the hex grid type. This change adds some logic to determine the proper rotation of the hexagonal patches.

No tests are added because I'm not sure how to test plotting?

## Why is the change being made?

Close #1878 .

---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [ ] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify any new/changed code.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/developer/tooling.html#add-release-notes) have been updated if necessary.
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

**Comment by keckler** [comment]
I am currently generating some plots to confirm that this is working.

**Comment by keckler** [comment]
Here is a correct corners-up core map that was generated with this branch:
image

Compare that against the pictures that were in the original issue: #1878 .

**Comment by keckler** [comment]
Here is the anl-acfi-177 facemap from [our publicly available blueprints](https://terrapower.github.io/armi/tutorials/walkthrough_inputs.html), which is flats-up:
image

So that confirms that flats-up are still working as expected.

File: doc/release/0.4.rst [hunk 1]
```diff
@@ -35,6 +35,7 @@ Bug Fixes
 #. Fixed ``PermissionError`` when using ``syncDbAfterWrite``. (`PR#1857 <https://github.com/terrapower/armi/pull/1857>`_)
 #. Fixed ``MpiDirectoryChanger``. (`PR#1853 <https://github.com/terrapower/armi/pull/1853>`_)
 #. Changed data type of ``thKernel`` setting from ``bool`` to ``str`` in ``ThermalHydraulicsPlugin``. (`PR#1855 <https://github.com/terrapower/armi/pull/1855>`_)
+#. Rotate hexaonal assembly patches correctly on facemap plots (`PR#1883 <https://github.com/terrapower/armi/pull/1883>`_)
```
Comment (john-science) [inline doc/release/0.4.rst:38]:
```suggestion
#. Rotate hexagonal assembly patches correctly on facemap plots. (`PR#1883 <https://github.com/terrapower/armi/pull/1883>`_)
```

**Review by john-science — APPROVED** [review]
Love it!

You're right about it being hard to unit test the plots.

I suppose you touched the `_makeAssemPatches()` function, which returns matplotlib objects. So we could perhaps add a unit test that covers that method by just inspecting the returned objects. I can see that test would have to be spun up from scratch, of course.

**Comment by keckler** [comment]
Yes, you're right, that could be a potential avenue for testing. I will try and put something up tomorrow. Thanks!

**Comment by keckler** [comment]
@john-science Take a look at the new test that I put up. If you like it, I believe this is ready to merge.

**Review by john-science — APPROVED** [review]
Awesome!

And thanks for the test!

**Pull request merged**